### ISSUE TITLE
Core: clean up PF utils

### ISSUE THREAD

**Issue by Rapster** [primefaces/primefaces#12519]
### Description

Based on https://github.com/primefaces/primefaces/pull/12512

Too many utils, not clear as to where to look when necessary

### Describe the solution you would like

IMO, we should stick to 3 utils at best:

- [ ] ComponentUtils _(no CompositeUtils, no TraversalUtils, no FacetUtils, the differences are very slight)_
- [ ] RendererUtils _(move a few CoreRenderer methods there)_
- [ ] ConverterUtils

A few methods can move elsewhere, for instance isFlex / isTouchable etc. can go in their respective classes

Also, 

- [ ] ~~MapBuilder should be  replaced by Map.of or Map.ofEntries~~ (this is tricky because Map.ofEntries does not allow NULL values)
- [x] PF class is useless, therefore should be deleted
- [x] Unused NestedSelectItem class
- [ ] ~LimitedSizeHashMap, still necessary?~ yes in use to defend against memory

**Comment by tandraschko** [comment]
i wont give input here the next ~5-6 weeks, will be in a big japan holiday soon and refactoring isnt high prio for me

**Comment by melloware** [comment]
@tandraschko have a great trip!  @Rapster i wouldn't mind waiting to do this until 16.0.X and would rather address some of the open bugs for 15.0.0 if we are putting our capacity somewhere.

**Comment by Rapster** [comment]
Should open bugs be fixed in 14.x and 15.x ? Don't see why waiting for 16 when 15 is far from being released

**Comment by melloware** [comment]
Yes they should I am just saying refactoring for 15.x for me is lower priority that addressing defects is all I am saying.

**Comment by pizzi80** [comment]
imho code cleanup and refactors are always good to ease bug fixing and improvements ... 

in java, with a decent IDE, refactors are easy and almost bug free

btw which is the release date of PF 15.0 ?

`LimitedSizeHashMap` could at least be renamed in `LRUMap`

**Comment by melloware** [comment]
No release date yet for 15.0.0

**Comment by Rapster** [comment]
> LimitedSizeHashMap could at least be renamed in LRUMap

HashMap is no LRUMap

> btw which is the release date of PF 15.0 ?

Wrong place to ask that question (open a discussion instead)

**Comment by pizzi80** [comment]
> > LimitedSizeHashMap could at least be renamed in LRUMap
> 
> HashMap is no LRUMap

the current PrimeFaces implementation it's backed by a `LinkedHashMap` 
and it's commonly called _LRUMap_ 
and it's more or less copy/pasted all over the internet ...

for example

https://github.com/eclipse-ee4j/mojarra/blob/master/impl/src/main/java/com/sun/faces/util/LRUMap.java

> 
> > btw which is the release date of PF 15.0 ?
> 
> Wrong place to ask that question (open a discussion instead)

:+1: 

... the main developers are all here ;)
... it was just to avoid opening a new thread ;)

**Comment by pizzi80** [comment]
> > LimitedSizeHashMap could at least be renamed in LRUMap
> 
> HashMap is no LRUMap

you're right this has no "access order" boolean flag set to true

**Comment by melloware** [comment]
@rapster will close for now. We have made some changes but the Utils to me help reduce one massive utility ComponentUtils.
